pkg/debug/debug.go

```diff
@@ -0,0 +1,144 @@
+package debug
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path"
+	"sort"
+	"strings"
+	"sync"
+
+	"github.com/spf13/afero"
+)
+
+const (
+	testArtifactsDirectoryVarName = "TEST_ARTIFACTS_DIRECTORY"
+)
+
+var (
+	// globals for overriding in unit tests
+	artifactsDirectoryBase = os.Getenv(testArtifactsDirectoryVarName) // nolint:gochecknoglobals
+	execCommand            = exec.Command                             // nolint:gochecknoglobals
+)
+
+// CollectArtifacts collects useful debugging artifacts from a given namespace.
+// Should typically be called if CurrentGinkgoTestDescription().Failed, like this:
+//   debug.CollectArtifacts(afero.NewOsFs(), GinkgoWriter, TestNamespace, KubeConfigPath, KubectlPath)
+func CollectArtifacts(fs afero.Fs, writer io.Writer, namespace, kubeConfigPath, kubectlPath string) {
```

Comment:
We should assume that a user of `test-tools` has a `client.Client` instance and make use of that here. This would become
```suggestion
func CollectArtifacts(client client.Client, fs afero.Fs, writer io.Writer, namespace string)
```

Reply:
See point 3 in https://github.com/kudobuilder/test-tools/pull/24#issuecomment-607104384 for reason why I'm using `kubectl`.

Reply:
This isn't about `kubectl`. `client.Client` is supposed to wrap everything necessary to use the various functions provided by this library. Collecting cluster artifacts should be one of those functions. Let's add `kubeconfigPath` as an optional value to the `client.Client` struct. This function will then use this value if it is set or return an error otherwise.